streams/src/main/java/org/apache/kafka/streams/kstream/Printed.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream;
+
+import org.apache.kafka.streams.errors.TopologyException;
+import org.apache.kafka.streams.kstream.internals.KStreamPrint;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+/**
+ * An object to define the options used when printing a {@link KStream}.
+ *
+ * @param <K> key type
+ * @param <V> value type
+ * @see KStream#print(Printed)
+ */
+public class Printed<K, V> {
+    private final PrintWriter printWriter;
+    private String label;
+    private KeyValueMapper<? super K, ? super V, String> mapper = new KeyValueMapper<K, V, String>() {
+        @Override
+        public String apply(final K key, final V value) {
+            return String.format("%s, %s", key, value);
+        }
+    };
+
+    private Printed(final PrintWriter printWriter) {
+        this.printWriter = printWriter;
+    }
+
+    /**
+     * Builds the {@link ProcessorSupplier} that will be used to print the records flowing through a {@link KStream}.
+     *
+     * @return the {@code ProcessorSupplier} to be used for printing
+     */
+    public ProcessorSupplier<K, V> build(final String processorName) {
```

Comment:
Could we not expose this public API?

Reply:
Yes, but then we'd need to expose all the other fields. Unless we can think of a way to work around this.

Reply:
Could we expose the getters in the sub class, or the build function in the sub class as you mentioned in the other PR?

Reply:
yeah we can.